src/adapters/scene_category.js

```diff
@@ -1,21 +1,66 @@
-import { Marker } from 'mapbox-gl--ENV';
 import constants from '../../config/constants.yml';
-import { createIcon } from '../adapters/icon_manager';
 import Telemetry from 'src/libs/telemetry';
 import { toUrl } from 'src/libs/pois';
 import { fire, listen } from 'src/libs/customEvents';
+import { poisToGeoJSON, emptyFeatureCollection } from 'src/libs/geojson';
+import { filteredPoisStyle, hoveredPoiStyle } from 'src/adapters/pois_styles';
+import { createDefaultMapIcon } from 'src/adapters/icon_manager';
 
-function getMarkerId(poi) {
-  return `marker_${poi.id}`;
-}
+const DYNAMIC_POIS_LAYER = 'poi-filtered';
+const ACTIVE_POIS_LAYER = 'poi-active';
 
 export default class SceneCategory {
   constructor(map) {
     this.map = map;
-    this.markers = [];
+
+    createDefaultMapIcon().then(imageData => {
+      this.map.addImage('pin_with_dot', imageData);
+    });
+
+    this.map.addSource(DYNAMIC_POIS_LAYER, {
+      type: 'geojson',
+      data: emptyFeatureCollection,
+      promoteId: 'id', // tells MapBox-GL to use this property as internal feature identifier
+    });
+    this.map.addLayer({
+      ...filteredPoisStyle,
+      source: DYNAMIC_POIS_LAYER,
+      id: DYNAMIC_POIS_LAYER,
+    });
+    this.map.on('click', DYNAMIC_POIS_LAYER, e => {
+      e.cancelMapClick = true; // Hack as MapBox events don't have stopPropagation
```

Comment:
```suggestion
      e.originalEvent.stopPropagation()
```

Reply:
See relevant suggestion here : https://github.com/QwantResearch/erdapfel/pull/773/files#r485606896 :)

Reply:
We can rely on the `originalEvent` property to set `cancelBubble` to `true` using `stopPropagation()`.
It just implies to check for `e.originalEvent.cancelBubble` where we don't want the mapbox event to be propagated.